### ISSUE TITLE
[QUESTION]: Classfical cross validation features

### ISSUE THREAD

**Issue by Keyan719** [chemprop/chemprop#279]
**What are you trying to do?**

I am sorry if this question has been previously asked. I am trying to using the cross validation function of chemprop train to produce a leave-one-out cross validation. However, I noticed a  issue (https://github.com/chemprop/chemprop/issues/41)  stating that chemprop's cross validation did not use the traditional k-fold method but instead perform k times shuffling I wonder if the traditional cv method has been added to chemprop. If so, which split type shall I refer to. If not, I guess I am supposed to use the 'predetermined' features. In this case, might I have a format reference about how the index file and directory shall look like? Thanks so much!

**Comment by cjmcgill** [comment]
You can use `--split_type cv` to do a traditional k-fold cross validation.

**Comment by Keyan719** [comment]
Thanks so much for replying! I am training the model with relatively small dataset so I am considering leave one out to utilize the most of the data. I still could not manage to perform leave one out cross validation, though. First, I tried --split_type cv-no-test and --num_folds as the size of my training set, and I got index out of range error (I attached the error log file here. My best guess is that I shall adjust --split_size accordingly?). And I noticed that the algorithm assigned two datapoints to the validation set instead of one. Could you give me some suggestion for this? Thanks so much
[log.txt](https://github.com/chemprop/chemprop/files/8693994/log.txt)

**Comment by cjmcgill** [comment]
There is a known issue with cv-no-test right now (#260). The fix for that has been delayed but will still happen. In the meantime, I think that `--split_type cv` should still work for you.

As a stopgap until cv-no-test is properly fixed. I _**think**_ that if you provide a `--separate_test_path {path/to/data.csv}` to some data that is valid when you select cv-no-test, it may bypass the error. This is an untested solution but I think it might work for now. If it doesn't work, then I would just suggest that you use `cv`.

I am not immediately sure why you are getting an extra datapoint in your Val set. Will take a look.

**Comment by Keyan719** [comment]
Thanks so much! Adding a separate test path works. For the 2 data point, I guess there might be a pseudo 1 added for validation set split, since when I set --num_folds as the dataset size, the validaiton set is 2 (1+1) and when I set -num_folds as half of the dataset size, the validation set contains three datapoints (2+1).

**Comment by cjmcgill** [comment]
Hi, #284 should have fixed some of your problems. If there are others still, please feel free to reopen this issue.
